models/Author.js

```diff
@@ -1,54 +1,66 @@
-'use strict';
+import mongoose from 'mongoose';
+import mockgoose from 'mockgoose';
 
-var mongoose = require('mongoose');
-var mongodb = require('common/makeCredential')('mongodb');
+if (process.env.NODE_ENV === 'test') {
+  mockgoose(mongoose);
+}
 
-var db = mongoose.createConnection(mongodb);
+const mongodbCredential = process.env.KINDLIZED_MONGODB;
+const db = mongoose.createConnection(mongodbCredential);
 
-var AuthorSchema = new mongoose.Schema({
+const AuthorSchema = new mongoose.Schema({
   name: {
     type: String,
     index: {
-      unique: true
-    }
+      unique: true,
+    },
   },
   lastModified: Date,
-  pageId: Number
+  pageId: Number,
 });
 
-var SequenceSchema = new mongoose.Schema({
+const SequenceSchema = new mongoose.Schema({
   name: String,
-  seq: Number
+  seq: Number,
 });
 
 SequenceSchema.index = { name: 1 };
-var SequenceModel = db.model('Sequence', SequenceSchema );
+const SequenceModel = db.model('Sequence', SequenceSchema );
 
-AuthorSchema.pre('save', function (next) {
-  var author = this;
-  var query, options, update;
-  if(!author.isNew){
+AuthorSchema.pre('save', function preSave(next) {
+  const author = this;
+  if (!author.isNew) {
     return next();
   }
-  query = {
-  };
-  update = {
-    $inc: {
-      seq: 1
+  const query = { name: 'sequencialID' };
+  const update = { $inc: { seq: 1 } };
+  const options = { upsert: true, new: true };
+
+  SequenceModel.find({}, (err, sequencialIDs)=> {
+    if (err) {
+      return next(err);
+    }
+    if (sequencialIDs.length > 0) {
+      return SequenceModel.findOneAndUpdate(query, update, options, (updateError, updatedSequencialID)=> {
+        if (updateError) {
+          return next(updateError);
+        }
+        author.pageId = updatedSequencialID.seq;
+        next();
+      });
     }
-  };
-  options = {
-    upsert: true
-  };
-  return SequenceModel.findOneAndUpdate(query, update, options, (function(_this) {
-    return function(err, data) {
-      if (!err && data) {
-        _this.pageId = data.seq;
-        return next();
+    const sequencialID = new SequenceModel({
```

Comment:
__[jshint]__
*  'const' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).